pkg/resources/manager/manager.go

```diff
@@ -202,24 +164,47 @@ func (r *resourceManager) initialize(ctx context.Context, machines []*api.Machin
 
 	r.ctx = ctx
 
-	err := r.loadTotalResources()
-	if err != nil {
-		return err
+	// Initialize all sources and check for common resources
+	var initializedResources *set.Set
+	for _, s := range r.sources {
+		resources, err := s.Init(r.ctx)
+		if err != nil {
+			return err
+		}
+
+		if initializedResources == nil {
+			initializedResources = resources
+		} else {
+			// Check for common resources with previously initialized sources
+			if initializedResources.Intersection(resources).Len() > 0 {
+				return ErrCommonResources
+			}
+
+			initializedResources = initializedResources.Union(resources)
+		}
 	}
 
-	//TODO implement "limiters"
+	r.log.Info("Initialized resources: " + r.convertResourcesToString(r.getAvailableResources()))
 
-	err = r.calculateAvailableResources(machines)
-	if err != nil {
-		return err
+	for _, machine := range machines {
```

Comment:
add here maybe comments what it is doing or put it into function.

Reply:
Comment added.